### PR TITLE
Update BlockItemMixin_accurateBlockPlacement.java

### PR DESCRIPTION
**Pull request by forestbat**
https://github.com/gnembon/carpe ... BlockPlacement.java
https://github.com/Virtuoel/Towelette/blob/master/src/main/java/virtuoel/towelette/mixin/BlockItemMixin.java

Your and Towelette's mixin have same redirect and made some mistakes,so I think this is more safe.

### CODE REVIEW

**Comment by DeadlyMC** [comment]
Looks like Virtuoel already fixed the conflict in this commit :
https://github.com/Virtuoel/Towelette/commit/b11256e81ae0466bd2d05f91a39e20ffa5b02837

**Comment by forestbat** [comment]
> 
> 
> Looks like Virtuoel already fixed the conflict in this commit :
> [Virtuoel/Towelette@b11256e](https://github.com/Virtuoel/Towelette/commit/b11256e81ae0466bd2d05f91a39e20ffa5b02837)

I saw this，however fabric don't recommend to use redirect attribute in mods，so the change is still useable。

**Comment by DeadlyMC** [comment]
I really don't get how your mixin works, what function from another mixin are your overriding in this case. And how is it retaining the same functionality as before

File: src/main/java/carpetextra/mixins/BlockItemMixin_accurateBlockPlacement.java [hunk 1]
```diff
@@ -13,11 +13,11 @@
 @Mixin(BlockItem.class)
 public class BlockItemMixin_accurateBlockPlacement
 {
-    @Redirect(method = "getPlacementState", at = @At(
-            value = "INVOKE",
-            target = "Lnet/minecraft/block/Block;getPlacementState(Lnet/minecraft/item/ItemPlacementContext;)Lnet/minecraft/block/BlockState;"
-    ))
-    private BlockState getAlternatePlacement(Block block, ItemPlacementContext itemPlacementContext_1)
+    @Shadow
+    private Block block //It's deprecated,but it's convenience
```
Comment (DeadlyMC) [inline src/main/java/carpetextra/mixins/BlockItemMixin_accurateBlockPlacement.java]: You are missing a `;` here, also i believe it needs to be `@Shadow @Final @Deprecated private Block block;`

**Comment by forestbat** [comment]
> 
> 
> I really don't get how your mixin works, what function from another mixin are your overriding in this case. And how is it retaining the same functionality as before

SoftOverride attribute is used to override method in class which is “mixined” by Mixin attribute，so I think it's effictive，if this can't solve problem，you can close this.

**Comment by DeadlyMC** [comment]
> > I really don't get how your mixin works, what function from another mixin are your overriding in this case. And how is it retaining the same functionality as before
> 
> SoftOverride attribute is used to override method in class which is “mixined” by Mixin attribute，so I think it's effictive，if this can't solve problem，you can close this.

Which method is that in this case?

**Comment by gnembon** [comment]
Double Injection in the same position can have worse sideeffects (i.e. undefined behaviour), than two redirects, which will prevent mods from running. This already saved us once in colliding with masa's minihud mod. I am not saying that Redirects are better - in that case thought we have agreed to use different if cases to not to hop on each other. 

I wouldn't close that for now, just learn more about it.

**Pull request closed without merging**